### PR TITLE
Invert Dependency on JavaScriptBridge

### PR DESCRIPTION
**Pull request by AnthonyMDev**
JavaScriptObjects now retain JavaScriptBridge
JavaScriptBridge `wrapperMap` now has weak references to `JavaScriptObject`s

Rename `ApolloCodgenFrontend` to `GraphQLJSFrontend`

I kept the JavaScript frontend as `ApolloCodgenFrontend` because it's not just `graphql-js`. It's a wrapper for graphql-js that specifically exposes the necessary pieces for the Apollo Codegen and has a few functions that handle parsing the graphls-js return results into the compilation result.

### CODE REVIEW

**Comment by calvincestari** [comment]
> I kept the JavaScript frontend as ApolloCodgenFrontend because it's not just graphql-js.

Do you mean the bundle? Looks like the class was [renamed](https://github.com/apollographql/apollo-ios/pull/2056/files#diff-d3c9d6c2762c2fda075e54e92fee8ff9f3a44422c6fbefbcdc56de00e5ee5522R104).

File: Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift [hunk 1]
```diff
@@ -266,9 +266,11 @@ public struct ApolloSchemaDownloader {
     """
   
   
-  static func downloadViaIntrospection(from endpointURL: URL,
-                                       httpMethod: ApolloSchemaDownloadConfiguration.DownloadMethod.HTTPMethod,
-                                       configuration: ApolloSchemaDownloadConfiguration) throws {
+  static func downloadViaIntrospection(
```
Comment (calvincestari) [inline Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift:269]: These are mostly formatting changes right - am I missing anything else?

File: Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const options = {
     file: "../dist/ApolloCodegenFrontend.bundle.js",
     format: "iife",
     name: "ApolloCodegenFrontend",
-    sourcemap: true,
+    sourcemap: false,
```
Comment (calvincestari) [inline Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: Does this remove the `ApolloCodegenFrontend.bundle.js.map` file? It looks like the file is still in the `dist` directory but empty. 😕

**Comment by AnthonyMDev** [comment]
I mean the `bundle.js` file. :)

File: Sources/ApolloCodegenLib/Frontend/JavaScriptBridge.swift [hunk 3]
```diff
@@ -126,15 +153,19 @@ class JavaScriptBridge {
...
-  // all objects that are passed over the bridge both on the Swift side and in JavaScript
-  // ('JSValue` is an Objective-C object that uses `JSValueProtect` to mark the underlying JavaScript
-  // object as inelligable for garbage collection.)
-  private var wrapperMap: [JSValue: JavaScriptObject] = [:]
+  /// We keep a map between `JSValue` objects and wrapper objects, to avoid repeatedly creating new
+  /// wrappers and to guarantee referential equality.
+  /// TODO: We may want to consider making the keys weak here, because this does mean we'll be
+  /// keeping alive all objects that are passed over the bridge in JavaScript.
+  /// ('JSValue` is an Objective-C object that uses `JSValueProtect` to mark the underlying
+  /// JavaScript object as inelligible for garbage collection.)
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/Frontend/JavaScriptBridge.swift:161]:
```suggestion
  /// JavaScript object as ineligible for garbage collection.)
```

File: Sources/ApolloCodegenLib/Frontend/dist/ApolloCodegenFrontend.bundle.js [hunk 4]
```diff
@@ -1,2 +1 @@
...
-//# sourceMappingURL=ApolloCodegenFrontend.bundle.js.map
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/Frontend/dist/ApolloCodegenFrontend.bundle.js:2]: Why is this no longer necessary? And why did the bundle map get deleted?

Comment (AnthonyMDev) [reply Sources/ApolloCodegenLib/Frontend/dist/ApolloCodegenFrontend.bundle.js:2]: It was never necessary, this is just cleanup of legacy. I changed the rollup config that creates the bundled js to remove the source mapping. As far as I can tell. that is only used for TypeScript libraries, and ours is pure JS.

File: Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js (hunk 2 above)
Comment (designatednerd) [reply Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: ^that explains the answer to one of my questions, but now I have a new one: Why are we turning this off?

File: Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift (hunk 1 above)
Comment (AnthonyMDev) [reply Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift:269]: Yupp, just cleaning up the formatting here.

File: Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js (hunk 2 above)
Comment (calvincestari) [reply Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: Do we need a source map? It's possible `true` is the default for that value in Rollup.

Comment (AnthonyMDev) [reply Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: Yeah that's what it does. The file was always empty, which is why I'm removing it. You're saying it's not being removed? My commit shows it deleted.
image

Comment (calvincestari) [reply Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: Yes you're correct, it is being deleted. I didn't look at the change properly. 👍🏻

Comment (calvincestari) [reply Sources/ApolloCodegenLib/Frontend/JavaScript/rollup.config.js:13]: > The file was always empty

It had content in before this deletion. I don't believe it was ever used though.

**Review by calvincestari — APPROVED** [review]

**Pull request merged**